server/src/main/java/io/crate/planner/operators/Rename.java

```diff
@@ -111,6 +111,9 @@ public LogicalPlan pruneOutputsExcept(SequencedCollection<Symbol> outputsToKeep)
         for (Symbol outputToKeep : outputsToKeep) {
             Symbols.intersection(outputToKeep, outputs, s -> {
                 Symbol childSymbol = parentToChildMap.get(s);
+                if (childSymbol == null) {
```

Comment:
If two equivalent symbols `(b >1)` with two different references occur, the outputs between rename and source might differ. Since the maps are identity hashmaps the only option I see is to check both variants.